src/cascadia/TerminalCore/terminalrenderdata.cpp

```diff
@@ -188,6 +206,23 @@ void Terminal::SelectNewRegion(const til::point coordStart, const til::point coo
     SetSelectionEnd(realCoordEnd, SelectionExpansion::Char);
 }
 
+void Terminal::SelectSearchRegions(std::vector<til::inclusive_rect> rects)
+{
+    _searchSelections.clear();
+    for (auto& rect : rects)
+    {
+        rect.top -= _VisibleStartIndex();
+        rect.bottom -= _VisibleStartIndex();
+
+        const auto realStart = _ConvertToBufferCell(til::point{ rect.left, rect.top });
+        const auto realEnd = _ConvertToBufferCell(til::point{ rect.right, rect.bottom });
+
+        auto rr = til::inclusive_rect{ realStart.x, realStart.y, realEnd.x, realEnd.y };
+
+        _searchSelections.emplace_back(rr);
+    }

```

Comment:
I was expecting to find something here poking the renderer.. like `TriggerSelection`. Is that already happening because we're also making a new selection when we search?

I am worried about the hidden coupling here.

Reply:
I think right now it is relying on this call to trigger selection.  https://github.com/microsoft/terminal/blob/9b823b03e47e9fbcaf321761123cf22f0ee7bcdd/src/cascadia/TerminalControl/ControlCore.cpp#L1624

Something does feel weird when I compare it to have the normal selection from _search.SelectCurrent() works.

Would it make sense to move the _searcher.HighlightSelections() here?  https://github.com/microsoft/terminal/blob/9b823b03e47e9fbcaf321761123cf22f0ee7bcdd/src/cascadia/TerminalControl/ControlCore.cpp#L1622

Or to add a TriggerSelection call here?  https://github.com/microsoft/terminal/blob/9b823b03e47e9fbcaf321761123cf22f0ee7bcdd/src/cascadia/TerminalControl/ControlCore.cpp#L1608

Reply:
That's a really good question. I think that you'll want to do it any time you are done changing `_searchSelections`. @lhecker might have a better opinion here.

I tried out your branch! It works really, really well. Congrats!

I only found one issue.

When I delete the search with backspace, or close it, the highlights stick around... but the search is no longer active. So next time the screen updates, they all go away.

I think this might be because of the missing `TriggerSelection`!

Reply:
somewhat long story on the selections not getting cleared when the last char is deleted from the search box.  I was trying to match the behavior from the release that I had on my machine and it looked like "current" selection highlight stuck around after deleting the last char and then I cross referenced visual studio and it looked like it left the highlights also so I thought it might be a pattern.

For clearing all selections, I think this will work.
https://github.com/e82eric/terminal/commit/4dc6305455d06c3cf6d05da08ccec2d6f32151ed

For leaving the current selection highlighted but clearing all of the other selections,  adding a TriggerSelection here would work.
https://github.com/e82eric/terminal/blob/4dc6305455d06c3cf6d05da08ccec2d6f32151ed/src/cascadia/TerminalControl/ControlCore.cpp#L1608

For clearing the selections after the search box is closed,  I actually like the behavior where it leaves the selections highlighted,  It gives me context if I want to go into mark mode about what is going to get selected.

Reply:
That all makes sense!